src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/MySqlTranslator.java

```diff
@@ -275,6 +275,34 @@ public String translate(DbColumn c) {
         }
     }
 
+    @Override
+    public String translateToCast(DbColumn dc) {
```

Comment:
I believe we should have a dedicated method to translate a `DbColumnType` to a string.  
This would be completely independent of the `CAST`.

This way you can have all `impl/*Translator.java`s to define how the types are translated.
And then you would have a very simple `translate(Cast) method.